kernel/posix/pthread_semaphore.c

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <pthread.h>
+
+int errno;
```

Comment:
Why is `errno` being defined here? Please `#include <errno.h>`; you'll need it anyway for the error codes.